src/screens/PhotosGalleryScreen/index.android.tsx

```diff
@@ -0,0 +1,238 @@
+    const photosProgressUpdateSubscription = DeviceEventEmitter.addListener('PHOTOS_PROGRESS_UPDATE', (update) => {
+      setPhotosProcessState(update);
+    });
+
+    const pressedItemSubscription = DeviceEventEmitter.addListener('PHOTOS_ITEM_PRESS', (pressedItem) => {
+      internxtMobilePhotosSdk
+        .getPhotosItem(pressedItem.name, pressedItem.type)
+        .then((item) => {
+          const photosItem: PhotosItem = {
+            photoId: item.photoId ?? null,
+            photoFileId: item.fileId,
+            previewFileId: item.previewId,
+            name: item.name,
+            takenAt: new Date(item.takenAt).getTime(),
+            updatedAt: new Date(item.updatedAt).getTime(),
+            width: item.width,
+            height: item.height,
+            format: item.type,
+            type: item.itemType === 'IMAGE' ? PhotosItemType.PHOTO : PhotosItemType.VIDEO,
+            localPreviewPath: pressedItem.previewPath ?? '',
+            localFullSizePath: pressedItem.previewPath ?? '',
+            status: PhotoSyncStatus.DEVICE_AND_IN_SYNC,
+            localUri: null,
+            bucketId: item.bucketId,
+            getSize: async () => {
+              return item.size;
+            },
+            getDisplayName: function (): string {
+              return `${item.name}.${item.type}`;
+            },
+          };
+          navigation.navigate('PhotosPreview', {
+            photosItem,
+          });
+        })
+        .catch((err) => {
+          errorService.reportError(err);
+          logger.error(`Failed to get photo item when receiving press event: ${JSON.stringify(err)}`);
+        });
+    });
+
+    return () => {
+      photosProgressUpdateSubscription.remove();
+      pressedItemSubscription.remove();
+    };
+  }, []);
+
+  useEffect(() => {
+    if (
+      (powerState as any) === 'charging' ||
+      (powerState as any) === 'full' ||
+      powerState.batteryState === 'charging' ||
+      powerState.batteryState === 'full'
+    ) {
+      setBackupsPausedDueToMissingConstraints(false);
+    } else {
+      setBackupsPausedDueToMissingConstraints(true);
+    }
+  }, [powerState]);
+
+  useEffect(() => {
+    photos.analytics.screen(PhotosAnalyticsScreenKey.PhotosGallery, { permissions: true });
+    return () => {
+      photosCtx.selection.resetSelectionMode();
+    };
+  }, []);
+
+  useEffect(() => {
+    if (photosCtx.syncEnabled) {
+      initPhotosMobileSdk();
+    } else {
+      stopPhotosMobileSdk();
+    }
+  }, [photosCtx.syncEnabled]);
+
+  const initPhotosMobileSdk = async () => {
+    try {
+      if (!photosCtx.syncEnabled) {
+        return logger.info('Photos sync is disabled');
+      }
+      const { user, device } = await photos.user.init();
+
+      await internxtMobilePhotosSdk.initialize({
+        photosBucketId: user.bucketId,
+        photosDeviceId: device.id,
+        photosUserId: user.id,
+      });
+
+      await internxtMobilePhotosSdk.startPhotos();
```

Comment:
Are we initializing the SDK only?

Reply:
WDYM? the Mobile SDK is initialized here: https://github.com/internxt/drive-mobile/blob/8f19c75c484754131f671b17ea23f564bdd58af1/src/store/slices/auth/index.ts#L31

This method only starts the native photos processor

Reply:
So, the method is called initPhotosMobileSdk but it does init the SDK (L141) and also does init the native photos processor, so the function naming is hiding a part of the behaviour this function has, I think

Reply:
I see what you mean now, I will decouple those